components/snipcart/actions/create-discount/create-discount.mjs

```diff
@@ -0,0 +1,122 @@
+import app from "../../snipcart.app.mjs";
+
+export default {
+  key: "snipcart-create-discount",
+  name: "Create Discount",
+  description: "Create a new Discount. [See the documentation](https://docs.snipcart.com/v3/api-reference/discounts#post-discounts)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    app,
+    name: {
+      propDefinition: [
+        app,
+        "name",
+      ],
+    },
+    maxNumberOfUsages: {
+      propDefinition: [
+        app,
+        "maxNumberOfUsages",
+      ],
+    },
+    trigger: {
+      propDefinition: [
+        app,
+        "trigger",
+      ],
+      reloadProps: true,
+    },
+    code: {
+      propDefinition: [
+        app,
+        "code",
+      ],
+      disabled: true,
+      hidden: true,
+    },
+    totalToReach: {
+      propDefinition: [
+        app,
+        "totalToReach",
+      ],
+      disabled: true,
+      hidden: true,
+    },
+    type: {
+      propDefinition: [
+        app,
+        "type",
+      ],
+      reloadProps: true,
+    },
+    amount: {
+      propDefinition: [
+        app,
+        "amount",
+      ],
+      disabled: true,
+      hidden: true,
+    },
+    rate: {
+      propDefinition: [
+        app,
+        "rate",
+      ],
+      disabled: true,
+      hidden: true,
+    },
+  },
+  async additionalProps() {
+    const props = {};
+    if (this.trigger === "Code") {
+      props.code = {
+        type: "string",
+        label: "Code",
+        description: "Code for the discount",
+      };
+    }
+    if (this.trigger === "Total") {
+      props.totalToReach = {
+        type: "string",
+        label: "Total to Reach",
+        description: "Minimum amount required to activate the discount",
+      };
+    }
+    if (this.type === "FixedAmount") {
+      props.amount = {
+        type: "string",
+        label: "Amount",
+        description: "Discount amount. Required when discount type is `FixedAmount`",
+      };
+    }
+    if (this.type === "Rate") {
+      props.rate = {
+        type: "string",
+        label: "Rate",
+        description: "Discount percentage, i.e.: `10`. Required when discount type is `Rate`",
+      };
+    }
+    return props;
```

Comment:
Since `code`, `totalToReach`, `amount`, and `rate` are already declared in the props section, I think you can just update their "hidden" and "disabled" values. (Applies to `update-discount.mjs` as well).
```suggestion
  async additionalProps(existingProps) {
    const props = {};
    if (this.trigger === "Code") {
      existingProps.code.hidden = false;
      existingProps.code.disabled: false,
    }
    if (this.trigger === "Total") {
      existingProps.totalToReach.hidden = false;
      existingProps.totalToReach.disabled = false;
    }
    if (this.type === "FixedAmount") {
      existingProps.amount.hidden = false;
      existingProps.amount.disabled = false;
    }
    if (this.type === "Rate") {
      existingProps.rate.hidden = false;
      existingProps.rate.disabled = false;
    }
    return props;
```